internal/integration/telemetry_sdk_emitter.go

```diff
@@ -56,10 +57,8 @@ func TelemetryHarvesterWithMetricsURL(url string) TelemetryHarvesterOpt {
 }
 
 // TelemetryHarvesterWithHarvestPeriod sets harvest period.
-func TelemetryHarvesterWithHarvestPeriod(t time.Duration) TelemetryHarvesterOpt {
-	return func(config *telemetry.Config) {
-		config.HarvestPeriod = t
-	}
+func telemetryHarvesterZeroPeriod(config *telemetry.Config) {
+	config.HarvestPeriod = 0
```

Comment:
I guess we are setting it to zero to disable the periodic harvest, did we check the behaiour? I am worried that if that is zero a different default value is used

Reply:
Yes, it is not documented but the implementation seems clear. A period of 0 disables periodic harvesting:
https://github.com/newrelic/newrelic-telemetry-sdk-go/blob/9ff9c3035f5a4745f2f3df7ed1b13c6595542b40/telemetry/harvester.go#L96